src/tools/synth_node.rs

```diff
@@ -26,22 +27,20 @@ pub struct SyntheticNode {
 }
 
 impl SyntheticNode {
-    pub async fn new(config: pea2pea::Config) -> Self {
+    pub async fn new(config: &TestConfig) -> Self {
         let (sender, receiver) = mpsc::channel(SYNTH_NODE_QUEUE_DEPTH);
         let inner = InnerNode::new(config, sender).await;
-        inner.enable_handshake().await;
+        if config.synth_node_config.do_handshake {
+            inner.enable_handshake().await;
+        }
         inner.enable_reading().await;
         inner.enable_writing().await;
         Self { inner, receiver }
     }
 
-    /// Starts node bound to localhost and with default [pea2pea::Config]
-    pub async fn start() -> Result<Self, AddrParseError> {
-        let node_config = pea2pea::Config {
-            listener_ip: Some("127.0.0.1".parse()?),
-            ..Default::default()
-        };
-        Ok(Self::new(node_config).await)
+    /// Starts node
+    pub async fn start(config: &TestConfig) -> Self {
+        Self::new(config).await
```

Comment:
doesn't seem like this function is needed, it does the same thing as `new`; one of them needs to go